### PR TITLE
Reorg/repayments

### PR DESCRIPTION
**Pull request by ritik99**

### CODE REVIEW

File: contracts/Pool/Pool.sol [hunk 1]
```diff
@@ -307,7 +272,7 @@ contract Pool is Initializable, IPool {
         uint256 _noOfRepaymentIntervals = poolConstants.noOfRepaymentIntervals;
         uint256 _repaymentInterval = poolConstants.repaymentInterval;
         IPoolFactory _poolFactory = IPoolFactory(PoolFactory);
-        IRepayment(_poolFactory.repaymentImpl()).initializeRepayment(_noOfRepaymentIntervals, _noOfRepaymentIntervals.mul(_repaymentInterval));
+        IRepayment(_poolFactory.repaymentImpl()).initializeRepayment(_noOfRepaymentIntervals, _repaymentInterval, poolConstants.borrowRate, poolConstants.loanStartTime);
```
Comment (prateekreddy) [inline contracts/Pool/Pool.sol:291]: If these are not used in poolFactory,  then we can delete these variables.

File: contracts/Pool/Pool.sol [hunk 2]
```diff
@@ -557,57 +503,35 @@ contract Pool is Initializable, IPool {
...
-            )
-        );
-        if (_interest < _extraInterest) {
-            _interest = 0;
-        } else {
-            _interest = _interest.sub(_extraInterest);
-        }
+        uint256 _repaymentPeriodCovered = IRepayment(_poolFactory.repaymentImpl()).getRepaymentPeriodCovered(address(this));
+        uint256 _interestAccruedThisPeriod = ((block.timestamp).sub(_repaymentPeriodCovered)).mul(_interestPerPeriod);
+        uint256 _repaymentOverdue = IRepayment(_poolFactory.repaymentImpl()).getRepaymentOverdue(address(this));
```
Comment (prateekreddy) [inline contracts/Pool/Pool.sol:549]: Can we combine above call and this to a single fn call.

File: contracts/Pool/Pool.sol (hunk 1 above)
Comment (prateekreddy) [reply contracts/Pool/Pool.sol:309]: I meant poolConstants.

File: contracts/Pool/Pool.sol (hunk 2 above)
Comment (prateekreddy) [reply contracts/Pool/Pool.sol:567]: There are 2 instances of repayment contract calls na. Can we combine them into one ?

File: contracts/Pool/Pool.sol (hunk 1 above)
Comment (ritik99) [reply contracts/Pool/Pool.sol:309]: poolConstant vars are getting used in multiple places though, right?

Comment (prateekreddy) [reply contracts/Pool/Pool.sol:309]: ohh... ok. Then it's fine.

File: contracts/Pool/Pool.sol (hunk 2 above)
Comment (ritik99) [reply contracts/Pool/Pool.sol]: Pushed new commit for this. getInterestCalculationVars() returns the two required variables

**Pull request merged**